frame/lending/src/mocks_offchain/mod.rs

```diff
@@ -0,0 +1,535 @@
+	pub const MaxLiquidationBatchSize: u32 = 5;
+}
+
+parameter_types! {
+	pub static WeightToFee: Balance = 1;
+}
+impl WeightToFeePolynomial for WeightToFee {
+	type Balance = Balance;
+
+	fn polynomial() -> WeightToFeeCoefficients<Self::Balance> {
+		let one = WeightToFeeCoefficient {
+			degree: 1,
+			coeff_frac: Perbill::zero(),
+			coeff_integer: WEIGHT_TO_FEE.with(|v| *v.borrow()),
+			negative: false,
+		};
+		smallvec![one]
+	}
+}
+
+impl pallet_lending::Config for Runtime {
+	type Oracle = Oracle;
+	type VaultId = VaultId;
+	type Vault = Vault;
+	type Event = Event;
+	type NativeCurrency = Balances;
+	type MultiCurrency = Tokens;
+	type CurrencyFactory = LpTokenFactory;
+	type Liquidation = Liquidations;
+	type UnixTime = Timestamp;
+	type MaxMarketCount = MaxLendingCount;
+	type AuthorityId = AuthorityId;
+	type WeightInfo = ();
+	type LiquidationStrategyId = LiquidationStrategyId;
+	type PalletId = LendingPalletId;
+	type OracleMarketCreationStake = OracleMarketCreationStake;
+	type MaxLiquidationBatchSize = MaxLiquidationBatchSize;
+	type WeightToFee = WeightToFee;
+}
+
+/// Convenience function to set the price of an asset in [`pallet_oracle::Prices`].
+///
+/// Sets the price at the current `System::block_number()`.
+pub fn set_price(asset_id: CurrencyId, new_price: Balance) {
+	let price = Price { price: new_price, block: System::block_number() };
+	pallet_oracle::Prices::<Runtime>::insert(asset_id, price);
+}
+
+// Build genesis storage according to the mock runtime.
+pub fn new_test_ext() -> sp_io::TestExternalities {
+	let mut storage = frame_system::GenesisConfig::default().build_storage::<Runtime>().unwrap();
+	let balances = vec![(*ALICE, 1_000_000_000), (*BOB, 1_000_000_000), (*CHARLIE, 1_000_000_000)];
+
+	pallet_balances::GenesisConfig::<Runtime> { balances }
+		.assimilate_storage(&mut storage)
+		.unwrap();
+	pallet_lending::GenesisConfig {}
+		.assimilate_storage::<Runtime>(&mut storage)
+		.unwrap();
+	GenesisBuild::<Runtime>::assimilate_storage(
+		&pallet_liquidations::GenesisConfig {},
+		&mut storage,
+	)
+	.unwrap();
+
+	let mut ext = sp_io::TestExternalities::new(storage);
+	ext.execute_with(|| {
+		System::set_block_number(1);
+		Timestamp::set_timestamp(MILLISECS_PER_BLOCK);
+	});
+	ext
+}
+
+// BLOCK HELPERS
+
+pub fn process_block_with_execution(extrinsic: TestExtrinsic) {
+	let block_number = System::block_number()
+		.safe_add(&1)
+		.expect("Hit the numeric limit for block number");
+	let header = Header::new(
+		block_number,
+		H256::default(),
+		H256::default(),
+		[69u8; 32].into(),
+		Digest::default(),
+	);
+	Executive::initialize_block(&header);
+	Timestamp::set_timestamp(MILLISECS_PER_BLOCK * block_number);
+	System::set_block_number(block_number);
+	Executive::apply_extrinsic(extrinsic).unwrap().unwrap();
+}
+
+/// Processes the specified amount alls [`next_block()`] and then calls
+/// [`Lending::finalize`](OnFinalize::on_finalize).
+pub fn process_and_progress_blocks(blocks_to_process: usize) {
```

Comment:
You can use `composable_tests_helpers::test::block::process_and_progress_blocks` instead of redefining the function (I just merged this earlier today: https://github.com/ComposableFi/composable/blob/a4af564fe79f0ff485b607635add70ba6da25d12/frame/composable-tests-helpers/src/test/block.rs)

Reply:
Done for tests.rs and tests_offchain.rs